napari/_commands/_layers.py

```diff
@@ -0,0 +1,248 @@
+    -------
+    str
+        A string like "ndarray" or "dask".
+    """
+    if isinstance(data, list):
+        if len(data) == 0:
+            return "EMPTY"
+        else:
+            # Recursively get the type string of the zeroth level.
+            return _get_type_str(data[0])
+
+    if type(data) == da.Array:
+        # Special case this because otherwise data_type.__name__
+        # below would just return "Array".
+        return "dask"
+
+    # For class numpy.ndarray this returns "ndarray"
+    return type(data).__name__
+
+
+def _get_size_str(data) -> str:
+    """Return human readable size.
+
+    Parameters
+    ----------
+    data
+        Layer's data.
+
+    Returns
+    -------
+    str
+        A string size like "24.2G".
+    """
+    if isinstance(data, list):
+        nbytes = sum(level.nbytes for level in data)
+    else:
+        nbytes = data.nbytes
+    return naturalsize(nbytes, gnu=True)
+
+
+class ListLayersTable:
+    """Table showing information about each layer.
+
+    Parameters
+    ----------
+    layers : List[Layer]
+        The layers to list in the table.
+
+    Attributes
+    ----------
+    table : TextTable
+        Formats our table for printing.
+    """
+
+    def __init__(self, layers: List[Layer]):
+        self.layers = layers
+        self.table = RowTable(
+            [
+                "ID",
+                {"name": "NAME", "align": "left"},
+                "LAYER",
+                "DATA",
+                "LEVELS",
+                "SHAPE",
+                "TOTAL",
+            ]
+        )
+        for i, layer in enumerate(self.layers):
+            self._add_row(i, layer)
+
+    def _get_shape_str(self, data):
+        """Get shape string for the data.
+
+        Either "NONE" or a tuple like "(10, 100, 100)".
+        """
+        if isinstance(data, list):
+            if len(data) == 0:
+                return "NONE"  # Shape layer is empty list?
+            else:
+                return f"{data[0].shape}"  # Multi-scale
+        else:
+            return str(data.shape)
+
+    @staticmethod
+    def _get_num_levels(data) -> int:
+        """Get the number of levels of the data.
+
+        Parameters
+        ----------
+        data
+            Layer data.
+
+        Returns
+        -------
+        int
+            The number of levels of the data.
+        """
+        if isinstance(data, list):
+            return len(data)
+        return 1
+
+    def _add_row(self, layer_id: int, layer: Layer) -> int:
+        """Add row describing one layer.
+
+        Parameters
+        ----------
+        layer_id : int
+            The layer id (from view.cmd.layers).
+        layer : Layer
+            The layer itself.
+        """
+        layer_type = type(layer).__name__
+        num_levels = self._get_num_levels(layer.data)
+        shape_str = self._get_shape_str(layer.data)
+        data_type = _get_type_str(layer.data)
+        size_str = _get_size_str(layer.data)
```

Comment:
Given my other comments above I think these functions/ calculations should live on the layers. It would also give you a bit more flexibility deal with all the corner cases of the different layers. i.e. I don't think file should be worried about differences between Surface/ Shapes/ and Image.

Instead you could maybe put the layer specific stuff in something `layer.info` and get back a tuple of `(layer.name, layer.layer_type, layer.data_type, layer.num_levels, layer.shape, layer.size)` where those things can be defined in the base layer and then modified as needed in layers like `Surface` or `Shapes`.

Alternatively you could use the `layer.__str__` to make the layer row be nicely formatted and then just compose those rows into a table here, adding the `layer_id` as needed. At that point though you might want `viewer.layers.__str__` to do that composition for you and just print the nice table too.

We could still keep the `cmd` though if you need to have more complex functionality. Or you could maybe one day we might have a `layer.levels` object and `layer.levels.__str__` that then gives you everything you need.

How does all this sound?